buckets/redis/bucket_factory.go

```diff
@@ -164,6 +166,40 @@ func (bf *bucketFactory) reconnectToRedis(oldClient *redis.Client) {
 	}
 }
 
+func (bf *bucketFactory) handleConnectionFailure(oldClient *redis.Client) {
+	bf.Lock()
+	defer bf.Unlock()
+
+	if oldClient == bf.client && !bf.connectionNeedsResolution {
+		logging.Print("Attempting to establish new connection to redis")
+		bf.connectionNeedsResolution = true
+		go bf.establishNewConnectionToRedis(oldClient)
+	}
+}
+
+func (bf *bucketFactory) establishNewConnectionToRedis(oldClient *redis.Client) {
+	client := oldClient
+	numsTried := 0
+
+	for true {
+		numsTried++
+		bf.reconnectToRedis(client)
```

Comment:
we need the backoff somewhere here, right now we're retrying constantly in a loop, without any break